evalml/tests/component_tests/test_imputer.py

```diff
@@ -334,34 +335,72 @@ def test_imputer_multitype_with_one_bool(data_type, make_data_type):
+    assert_frame_equal(X_multi_expected_arr, X_multi_t)
+
+
+def test_imputer_int_preserved():
+    X = pd.DataFrame(pd.Series([1, 2, 11, np.nan]))
+    imputer = Imputer(numeric_impute_strategy="mean")
+    transformed = imputer.fit_transform(X)
+    pd.testing.assert_frame_equal(transformed, pd.DataFrame(pd.Series([1, 2, 11, 14 / 3])))
+    assert transformed.ww.logical_types == {0: Double}
+
+    X = pd.DataFrame(pd.Series([1, 2, 3, np.nan]))
+    imputer = Imputer(numeric_impute_strategy="mean")
+    transformed = imputer.fit_transform(X)
+    pd.testing.assert_frame_equal(transformed, pd.DataFrame(pd.Series([1, 2, 3, 2])), check_dtype=False)
+    assert transformed.ww.logical_types == {0: Double}
+
+    X = pd.DataFrame(pd.Series([1, 2, 3, 4], dtype='int'))
+    imputer = Imputer(numeric_impute_strategy="mean")
+    transformed = imputer.fit_transform(X)
+    pd.testing.assert_frame_equal(transformed, pd.DataFrame(pd.Series([1, 2, 3, 4])), check_dtype=False)
+    assert transformed.ww.logical_types == {0: Integer}
+
+
+def test_imputer_bool_preserved():
+    X = pd.DataFrame(pd.Series([True, False, True, np.nan]))
+    imputer = Imputer(categorical_impute_strategy="most_frequent")
+    transformed = imputer.fit_transform(X)
+    pd.testing.assert_frame_equal(transformed, pd.DataFrame(pd.Series([True, False, True, True], dtype="category")))
+    assert transformed.ww.logical_types == {0: Categorical}
+
+    X = pd.DataFrame(pd.Series([True, False, True, False]))
+    imputer = Imputer(categorical_impute_strategy="most_frequent")
+    transformed = imputer.fit_transform(X)
+    pd.testing.assert_frame_equal(transformed, pd.DataFrame(pd.Series([True, False, True, False])), check_dtype=False)
+    assert transformed.ww.logical_types == {0: Boolean}
 
 
 @pytest.mark.parametrize("X_df", [pd.DataFrame(pd.Series([1, 2, 3], dtype="Int64")),
                                   pd.DataFrame(pd.Series([1., 2., 4.], dtype="float")),
                                   pd.DataFrame(pd.Series(['a', 'b', 'a'], dtype="category")),
-                                  pd.DataFrame(pd.Series([True, False, True], dtype="boolean")),
+                                  pd.DataFrame(pd.Series([True, False, True], dtype=bool)),
                                   pd.DataFrame(pd.Series(['this will be a natural language column because length', 'yay', 'hay'], dtype="string"))])
 @pytest.mark.parametrize("has_nan", [True, False])
 @pytest.mark.parametrize("numeric_impute_strategy", ["mean", "median", "most_frequent"])
 def test_imputer_woodwork_custom_overrides_returned_by_components(X_df, has_nan, numeric_impute_strategy):
     y = pd.Series([1, 2, 1])
-    if has_nan:
-        X_df.iloc[len(X_df) - 1, 0] = np.nan
     override_types = [Integer, Double, Categorical, NaturalLanguage, Boolean]
     for logical_type in override_types:
+        # Column with Nans to boolean used to fail. Now it doesn't
+        if has_nan and logical_type == Boolean:
+            continue
```

Comment:
Wait if this case used to fail but now it doesn't, why are we `continue`ing here?

Reply:
The woodwork schema is invalid: https://github.com/alteryx/woodwork/issues/846

Reply:
Got it. In that case, perhaps the comment should say `Column with Nans to boolean used to fail. Now it doesn't, but it should` ?

Reply:
Will make that change!